prompt_toolkit/key_binding/bindings/emacs.py

```diff
@@ -139,6 +144,11 @@ def is_returnable() -> bool:
     handle('escape', 'enter', filter=insert_mode & is_returnable)(
         get_by_name('accept-line'))
 
+    # It appears that on windows, Enter -> ^M but Ctrl-Enter brings in a ^J;
+    # We handle this the same as escape-enter
+    handle('escape', 'c-j', filter=insert_mode & is_returnable)(
```

Comment:
This is probably due to a bug in ``input/win32.py``. It's supposed to be ``'escape', 'c-m'`` too.

Reply:
Yes, I wasn't too sure how to address this, and didn't want to break things by modifying `input/win32.py' incorrectly. There seems to be some inconsistencies in the code. The issue is that on windows,
* `enter` brings `'\r'` - ascii 13 - `'c-m'`
* `ctrl-enter` brings `'\n'` - ascii 10 - `'c-j'` and the `CTRL_PRESSED` flag

`input/win32.py` converts `CTRL_PRESSED 'c-j'` (which is `ctrl-enter`) into `'escape', 'c-j'`, but then that is NOT handled by the `'escape','enter'` binding, because `'enter'` is `'c-m'` **(issue 1)**. 

Also earlier in `input/win32.py`, we find this piece of code:
```
if self.mappings[ascii_char] == Keys.ControlJ:
    u_char = '\n'  # Windows sends \n, turn into \r for unix compatibility.
result = KeyPress(self.mappings[ascii_char], u_char)
```
The comment seems to tell the correct story, but then that is not what is done. `u_char` should be set to `'\r'` here (setting it to `'\n'` doesn't do anything; it is already that) and then we should call again `ascii_char = u_char.encode('utf-8')` so that in the next line (`result = ...`) we get `KeyPress('c-m', '\r')`. At the moment, hitting `ctrl-enter` still leaves us with a `'c-j'` **(issue 2)**. On top of that fix then, we have to fix issue 1 above: it should be  `CTRL_PRESSED 'c-m'` that is converted into `'escape', 'c-m'` rather than `CTRL_PRESSED 'c-j.'` 

I can see this would work, but then wasn't sure if this wouldn't break anything else. Given the importance of the `enter` key, I didn't want to mess it up. Should I fix that as part of this pull request, or should I make it a separate pull request?

Reply:
Have actually tried the solution I propose above, but for some reason it does not work. When hitting `ctrl-enter`, the prompt displays `^[` and then insert a carrier return (adds a line). Like the two keys, `'escape','c-m'`, are treated separately. Note that when I hit `ctrl-Q` (to display escape sequences) and then `ctrl-enter`, it does display `^[^M`, so the `c-j` is correctly converted to `c-m`, but somehow something else happens after that ...

Reply:
Maybe there is a problem with the KEY_ALIASES mechanism. Like the `'escape','c-m'` is not actually converted into `'escape','enter'` ... Just my 5c here, not too sure when this is called.

Reply:
Got a bit deeper in this issue here. The problem seems to be that `'escape','c-m'` is turned into a `BracketedPaste` event because `win32.ConsoleInputReader._is_paste` returns true for that key combination (`'escape'` is considered `str` and `'c-m'` is a newline.) Not sure what to do here. Might be simpler to pass `ctrl-enter` as `'escape','c-j'` and binds to that rather than tamper with the paste mechanism? Or even define a `'c-enter'` key, and not translate it into an `escape' combination at all?

Reply:
Right. Thanks for taking the time to dig into this issue.
I'm not sure yet what's the best approach. The ``_is_paste`` method is a workaround, because the Windows console doesn't support bracketed paste, but it's important to detect it.
Probably we can simply ignore the escape key in here, and it should be fine.

The chance that people literally paste "escape + enter" and expect it to be inserted as text literally like that is very small...